### DISCUSSION TITLE
Add me to solomon-data (or how to get the eaf script some other way?)

### DISCUSSION THREAD

**Discussion by sarpu** [LAAC-LSCP/ChildProject#205] [General]
I tried to access the eaf script to update it, but I don't think I have permission to view the solomon-data repo. This is the link:
https://github.com/LAAC-LSCP/solomon-data/blob/master/scripts/import_eaf.py

**Comment by lucasgautheron** [comment]
Oops sorry, here it is:

```python
#!/usr/bin/env python3
from ChildProject.projects import ChildProject
from ChildProject.annotations import AnnotationManager
import os
import pandas as pd
import glob
import pympi

project = ChildProject('.')
am = AnnotationManager(project)

files = pd.DataFrame([
    {'raw_filename': os.path.basename(f)}
    for a in['BC', 'NK', 'NM', 'LD']
    for f in glob.glob(os.path.join('annotations/solis', a, 'raw/*.eaf'))
])

extract = files['raw_filename'].str.extract(r"([0-9]+_(?:CW[0-9]+|NA)_CH[0-9]+_(?:AJ|FB|LM)[0-9]+_(?:AJ|FB|LM)[0-9]+_[0-9]{6})_([0-9])?(?:.*)(?:\-|_)([A-Z]{2})")
files['recording_filename'] = extract[0] + '.WAV'
files['part'] = extract[1]
files['author'] = extract[2]
files = files.merge(project.recordings, how = 'inner', left_on = 'recording_filename', right_on = 'recording_filename')

files['part'] = files['part'].fillna(1)
files['time_seek'] = (files['part'].astype(int)-1)*15*3600*1000
files['set'] = files['author'].apply(lambda author: os.path.join('solis', author))

ranges = []
for f in files.to_dict(orient = 'records'):
    eaf = pympi.Elan.Eaf(os.path.join('annotations', f['set'], 'raw', f['raw_filename']))
    portions = eaf.tiers['code_periodic'][0]

    for pid in portions:
        (start_ts, end_ts, value, svg_ref) = portions[pid]
        (start_t, end_t) = (eaf.timeslots[start_ts], eaf.timeslots[end_ts])
        
        ranges.append({'raw_filename': f['raw_filename'], 'range_onset': int(start_t), 'range_offset': int(end_t)})

ranges = pd.DataFrame(ranges)

input = files[['recording_filename', 'raw_filename', 'time_seek', 'set']]
input['format'] = 'eaf'

ranges = ranges.merge(input, how = 'left', left_on = 'raw_filename', right_on = 'raw_filename')

am.remove_set('solis', recursive = True)

am.import_annotations(ranges)
```

**Reply by sarpu** [reply]
Thank you

**Comment by sarpu** [comment]
Actually, is it possible to get a sample eaf from the solomon dataset (or the dataset that this script was used to import)? It would be very very helpful to be able to step through the code seeing what it's actually doing in that way. I think I have a general idea, but since the eaf tier names, etc. don't seem to match, I feel like I could speed this  way up if I ad access to a sample file.

**Reply by sarpu** [reply]
For example, which tier in lucid corresponds to this line from the script above: `    portions = eaf.tiers['code_periodic'][0]`? Lucid eafs don't have `code_periodic`?

**Reply by lucasgautheron** [reply]
> Actually, is it possible to get a sample eaf from the solomon dataset (or the dataset that this script was used to import)? It would be very very helpful to be able to step through the code seeing what it's actually doing in that way. I think I have a general idea, but since the eaf tier names, etc. don't seem to match, I feel like I could speed this way up if I ad access to a sample file.

I just emailed you one file from solomon, I will send you more later

**Reply by lucasgautheron** [reply]
> For example, which tier in lucid corresponds to this line from the script above: ` portions = eaf.tiers['code_periodic'][0]`? Lucid eafs don't have `code_periodic`?

According to ACLEW DAS templates, the tiers should be `code_periodic` or `code_random` according to how the portions to annotate have been selected (see https://github.com/LAAC-LSCP/ChildProject/blob/master/ChildProject/pipelines/eafbuilder.py#L18)

Is there any `code_random` tier in lucid eafs ?

**Reply by sarpu** [reply]
Nope, these are the tiers:

```
['CHI', 'FA1', 'xds@FA1', 'lex@CHI', 'mwu@CHI', 'code', 'context', 'code_num', 'on_off', 'UC1', 'xds@UC1', 'MA1', 'xds@MA1', 'FA2', 'xds@FA2', 'FA3', 'xds@FA3', 'FA4', 'xds@FA4', 'EE1', 'xds@EE1', 'notes', 'vcm@CHI']
```
Maybe `code_num`?

**Reply by lucasgautheron** [reply]
Maybe `code_num` indeed, or `code` (do they return annotations with identical bounds?)

**Reply by sarpu** [reply]
No unfortunately
```
>>> eaf.tiers['code']
({'a2': ('ts2', 'ts146', 'random-1', None), 'a995': ('ts153', 'ts301', '11. TS 09/09/18', None), 'a6': ('ts306', 'ts422', 'random-2', None), 'a380': ('ts429', 'ts610', 'random-6', None), 'a368': ('ts617', 'ts726', 'random-7', None), 'a1003': ('ts733', 'ts795', '12. TS 12/08/18', None), 'a999': ('ts800', 'ts873', '13. TS 12/08/18', None), 'a1011': ('ts879', 'ts1011', '14. TS 15/08/18', None), 'a372': ('ts1016', 'ts1235', 'random-8', None), 'a1007': ('ts1243', 'ts1351', '15. TS 15/08/18', None), 'a364': ('ts1356', 'ts1359', 'random-9', None), 'a10': ('ts1364', 'ts1367', 'random-3', None), 'a376': ('ts1372', 'ts1375', 'random-10', None), 'a14': ('ts1380', 'ts1383', 'random-4', None), 'a18': ('ts1388', 'ts1391', 'random-5', None)}, {}, {'LINGUISTIC_TYPE_REF': 'transcription', 'TIER_ID': 'code'}, 5)
>>> eaf.tiers['code_num']
({'a3': ('ts3', 'ts147', '1', None), 'a996': ('ts154', 'ts302', '11', None), 'a7': ('ts307', 'ts423', '2', None), 'a381': ('ts430', 'ts611', '9', None), 'a369': ('ts618', 'ts727', '10', None), 'a1004': ('ts734', 'ts796', '14', None), 'a1000': ('ts801', 'ts874', '15', None), 'a1012': ('ts880', 'ts1012', '13', None), 'a373': ('ts1017', 'ts1236', '7', None), 'a1008': ('ts1244', 'ts1352', '12', None), 'a365': ('ts1357', 'ts1360', '6', None), 'a11': ('ts1365', 'ts1368', '3', None), 'a377': ('ts1373', 'ts1376', '8', None), 'a15': ('ts1381', 'ts1384', '4', None), 'a19': ('ts1389', 'ts1392', '5', None)}, {}, {'LINGUISTIC_TYPE_REF': 'transcription', 'TIER_ID': 'code_num'}, 7)
```
I'll just use one of those for now, and when we know the correct tier to use, it should be simple to just change it.

**Reply by lucasgautheron** [reply]
I had a look at one file and it seems to me they are equal
If you look at 'ts2' and 'ts3' [here](https://github.com/aclew/raw_LUC/blob/master/1156.eaf#L10) for instance, they refer to the same timestamp:

```xml
<TIME_SLOT TIME_SLOT_ID="ts2" TIME_VALUE="1980000"/>
<TIME_SLOT TIME_SLOT_ID="ts3" TIME_VALUE="1980000"/>
```

(same for ts146 and ts147)

**Reply by sarpu** [reply]
great catch, I didn't check the actual time values.

**Comment by sarpu** [comment]
Yep, it is solved.

Le jeu. 3 juin 2021 à 10:52, Lucas Gautheron ***@***.***> a
écrit :
